src/services/lessons/lessons.service.js

```diff
@@ -124,6 +121,18 @@ class Lessons {
 			throw new Forbidden(this.err.noAccess);
 		}
 
+		if (params.query.all === 'true') {
+			// call sections via route and not populate because of permission check and socket channels
+			const sections = await this.app.service('lesson/:lessonId/sections').find({
```

Comment:
In most situations the access is true, also the section request should not wait for lesson.